di_website/datasection/models.py

```diff
@@ -0,0 +1,86 @@
+from django.db import models
+
+import random
+
+from wagtail.admin.edit_handlers import (
+    FieldPanel,
+    InlinePanel,
+    PageChooserPanel,
+    StreamFieldPanel
+)
+from wagtail.core.fields import RichTextField, StreamField
+from wagtail.core.models import Page
+from wagtail.core.blocks import StreamBlock
+
+from di_website.common.base import hero_panels, get_related_pages
+from di_website.common.mixins import OtherPageMixin, HeroMixin, TypesetBodyMixin
+from di_website.common.constants import MAX_RELATED_LINKS
+
+from modelcluster.fields import ParentalKey
+
+from .blocks import QuoteStreamBlock, DataSuportStreamBlock
+
+
+class DataSectionPage(TypesetBodyMixin, HeroMixin, Page):
+    """ Main page for datasets """
+
+    quotes = StreamField(
+        QuoteStreamBlock,
+        verbose_name="Quotes",
+        null=True,
+        blank=True
+    )
+
+    dataset_subtitle = models.TextField(
+        null=False,
+        blank=False,
+        default="",
+        help_text='A description of the datasets'
+    )
+
+    data_support = StreamField(
+        DataSuportStreamBlock(max_num=1),
+        verbose_name="Data Support Services",
+        null=True,
+        blank=True
+    )
+
+    content_panels = Page.content_panels + [
+        hero_panels(),
+        StreamFieldPanel('body'),
+        StreamFieldPanel('quotes'),
+        FieldPanel('dataset_subtitle'),
+        StreamFieldPanel('data_support'),
+        InlinePanel('datasection_related_links',
+            label='More About Us',
+            max_num=MAX_RELATED_LINKS
+        )
+
+    ]
+
+    parent_page_types = ['home.HomePage']
+
+    class Meta:
+        verbose_name = "Data Section Page"
+
+    def getRandomQuote(self):
```

Comment:
I think this will only ever return either the first or the second quote. Could you query the length of `self.quotes` and use that length to pick a random quote from the full list? Also, please make sure this function always returns something. As written, if the length of `self.quotes` is 0, the for-loop will never trigger and the function will never return.

Reply:
Created a separate function to count number of quotes in a for loop; Also added a return statement at the end of `getRandomQuote()` and a check in template to hide html for when no quotes exist.